src/http/client.c

```diff
@@ -158,11 +158,11 @@ static void conn_idle(struct conn *conn)
 		return;
 
 	req =  conn->req;
-	if (req)
+	if (!req)
 		return;
```

Comment:
Like so:

```
static void conn_idle(struct conn *conn)
{
        struct http_req *req;
        struct http_cli *cli;
        if (!conn)
                goto out;

        req =  conn->req;
        if (!req)
                goto out;

        cli = req->cli;
        if (!cli)
                goto out;

        tmr_start(&conn->tmr, cli->conf.idle_timeout, timeout_handler, conn);

out:
        conn->req = NULL;
}
```

Reply:
You are right. Since conn_idle will return `if(!conn)`. It would lead to an assignment of a NULL ptr in req_close if req->conn is null. Removing `req->conn->req = NULL` and going for the out label prevents that.